File: src/main/java/com/bvanseg/gigeresque/client/entity/model/RunnerbursterEntityModel.java
package com.bvanseg.gigeresque.client.entity.model;

import com.bvanseg.gigeresque.client.entity.animation.EntityAnimations;
import com.bvanseg.gigeresque.client.entity.texture.EntityTextures;
import com.bvanseg.gigeresque.common.entity.impl.RunnerbursterEntity;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.util.Identifier;
import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
import software.bernie.geckolib3.model.AnimatedGeoModel;
import software.bernie.geckolib3.model.provider.data.EntityModelData;

import java.util.List;

@Environment(EnvType.CLIENT)
public class RunnerbursterEntityModel extends AnimatedGeoModel<RunnerbursterEntity> {
    @Override
    public Identifier getModelLocation(RunnerbursterEntity object) {
        return EntityModels.RUNNERBURSTER;
    }

    @Override
    public Identifier getTextureLocation(RunnerbursterEntity object) {
        return EntityTextures.RUNNERBURSTER;
    }

    @Override
    public Identifier getAnimationFileLocation(RunnerbursterEntity animatable) {
        return EntityAnimations.RUNNERBURSTER;
    }

    @Override
    public void setLivingAnimations(RunnerbursterEntity entity, Integer uniqueID, AnimationEvent customPredicate) {
        super.setLivingAnimations(entity, uniqueID, customPredicate);
        var neck = getAnimationProcessor().getBone("neck");
        List<EntityModelData> extraDataList = customPredicate.getExtraDataOfType(EntityModelData.class);
        if (extraDataList.isEmpty()) return;
        var extraData = extraDataList.get(0);
        neck.setRotationY(extraData.netHeadYaw * ((float) Math.PI) / 340f);
    }
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/model/RunnerbursterEntityModel.kt
package com.bvanseg.gigeresque.client.entity.model

import com.bvanseg.gigeresque.client.entity.animation.EntityAnimations
import com.bvanseg.gigeresque.client.entity.texture.EntityTextures
import com.bvanseg.gigeresque.common.entity.impl.RunnerbursterEntity
import net.minecraft.util.Identifier
import software.bernie.geckolib3.core.event.predicate.AnimationEvent
import software.bernie.geckolib3.model.AnimatedGeoModel
import software.bernie.geckolib3.model.provider.data.EntityModelData

/**
 * @author Boston Vanseghi
 */
class RunnerbursterEntityModel : AnimatedGeoModel<RunnerbursterEntity>() {
    override fun getModelLocation(entity: RunnerbursterEntity): Identifier = EntityModels.RUNNERBURSTER
    override fun getTextureLocation(entity: RunnerbursterEntity): Identifier = EntityTextures.RUNNERBURSTER
    override fun getAnimationFileLocation(animatable: RunnerbursterEntity): Identifier = EntityAnimations.RUNNERBURSTER

    override fun setLivingAnimations(entity: RunnerbursterEntity, uniqueID: Int, customPredicate: AnimationEvent<*>) {
        super.setLivingAnimations(entity, uniqueID, customPredicate)
        val neck = animationProcessor.getBone("neck")
        val extraData = customPredicate.getExtraDataOfType(EntityModelData::class.java).firstOrNull() ?: return
        neck.rotationY = extraData.netHeadYaw * (Math.PI.toFloat() / 340f)
    }
}